sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/image/image_search_space_utils.py

```diff
@@ -158,13 +165,43 @@ def _convert_sweep_dist_dict_to_str_dict(sweep_distribution: dict) -> dict:
     return sweep_distribution
 
 
+class ChoicePlusSchema(ChoiceSchema):
+    """ Choice schema that allows boolean values also"""
+    values = fields.List(
+        UnionField(
+            [
+                DumpableIntegerField(strict=True),
+                DumpableStringField(),
+                fields.Float(),
+                fields.Boolean(),
```

Comment:
shouldn't we just update ChoiceSchema to include boolean instead of a new class?

Reply:
Yes Riti. We are already doing that BooleanSearchSpace. The choicePlus schema is required while we do the intermediate conversion from SweepDistributionObj -> String for pipeline job.
At that point, we don't have the parameter information, so based on the SweepDistributionObj type, we are parsing them. Currently the issue is, the current choiceSchema does not have list of booleans and that's why it is required to add those.

Reply:
yes, I meant why not add boolean to current choiceSchema?

Reply:
Good Point Riti. The problem is ChoiceSchema is used for all other hyperdrive jobs as well. They currently donot support choice of booleans directly, so I had to create ChoicePlusSchema only for vision. 

It is working for us since we send everything to hyperdrive as string.